### DISCUSSION TITLE
Profile selection

### DISCUSSION THREAD

**Discussion by cryptocake** [davidrapan/ha-solarman#53] [General]
I'm trying to setup this integration (v24.07.24), but I'm a bit stuck on the profile selection. 
I have a **Solis eh3p10k-h-zp** and I'm not sure if any of the existing Solis profiles would match? It did auto select the `deye_hybrid`, but I'm not sure if that's the case or not.
My gut says it needs a custom profile, is that true? And if so: How would I create one?

Thank you.

**Comment by PV-Joe** [comment]
Did you try the Solis profiles and check if any of the sensors are working?
If not then you need the REGISTERS for the right values to create a profile.

Did you try this profile?
https://github.com/StephanJoubert/home_assistant_solarman/pull/627/commits/56374937fe2d289e29ed443e708e868d60db3f73

I'm using a DEYE Hybrid, so I won't be much help.
Good luck!

**Comment by davidrapan** [comment]
I can add the profile @PV-Joe just suggested for you.

Edit: [Solis 3P-5g](https://github.com/davidrapan/ha-solarman/blob/main/custom_components/solarman/inverter_definitions/solis_3p-5g.yaml)

**Comment by cryptocake** [comment]
Thank you. I am trying to get it connected with that profile, but it keeps giving me this error:
```
Logger: custom_components.solarman.api
Source: custom_components/solarman/api.py:167
integration: Solarman ([documentation](https://github.com/davidrapan/ha-solarman), [issues](https://github.com/davidrapan/ha-solarman/issues))
First occurred: 18:25:54 (11 occurrences)
Last logged: 18:28:55

Querying (2999 - 3043) failed. #0 [IllegalDataAddressError: The data address received in the request is not an allowable address for the server. ]
Querying (33029 - 33179) failed. #0 [IllegalDataValueError: The value contained in the request data field is not an allowable value for the server. ]
```

Is this a problem with the registers in the profile? Or something else?

**Reply by davidrapan** [reply]
Yes it looks like for some kind of an issue with registers... I would try to selectively pin point those which causes issues by trying to run the profile with just few of them and then try to add more and so.
